packages/main/src/plugin/color-registry.ts

```diff
@@ -27,12 +27,108 @@ import tailwindColorPalette from '../../../../tailwind-color-palette.json' with
 import { isWindows } from '../util.js';
 import type { ApiSenderType } from './api.js';
 import { AppearanceSettings } from './appearance-settings.js';
+import { colorDefinition, ColorDefinitionBuilder, colorPalette, ColorPaletteHelper } from './color-builder.js';
 import type { ConfigurationRegistry } from './configuration-registry.js';
 import { Disposable } from './types/disposable.js';
 
+export { colorDefinition, ColorDefinitionBuilder, colorPalette, ColorPaletteHelper };
+
 const { amber, black, charcoal, dustypurple, fuschia, gray, green, purple, red, sky, stone, white, transparent } =
   tailwindColorPalette;
 
+/**
+ * Apply alpha to a color string and return the formatted CSS color.
+ * @param color - The color string to apply alpha to
+ * @param alphaValue - The alpha value (0-1)
+ * @returns The formatted CSS color string
+ * @throws Error if the color cannot be parsed or formatted
+ */
+function applyAlpha(color: string, alphaValue: number): string {
+  if (alphaValue === 1) {
+    return color;
+  }
+
+  const parsed = parse(color);
+  if (!parsed) throw new Error(`Failed to parse color ${color}`);
+  parsed.alpha = alphaValue;
+
+  const formatted = formatCss(parsed);
+  if (!formatted) throw new Error(`Failed to format color ${color}`);
+  return formatted;
+}
+
+/**
+ * Builder class for fluent color definition creation.
+ * Does not register colors directly - call build() to get the color definition object,
+ * then pass it to registerColorDefinition().
+ * Supports optional alpha (opacity) values for each theme variant.
+ *
+ * @example
+ * this.registerColorDefinition(
+ *   this.color('my-color').withLight('#ffffff').withDark('#000000').build()
+ * );
+ *
+ * @example
+ * this.registerColorDefinition(
+ *   this.color('my-transparent-color').withLight('#ffffff', 0.5).withDark('#000000', 0.8).build()
+ * );
+ */
+export class ColorBuilder {
```

Comment:
remark(blocking): this PR introduces a new `color-builder.ts` file but this new class `ColorBuilder` is added to the existing `color-registry.ts` file and this does not make semantic sense to me.

Reply:
> Would you be okay deferring that refactor to keep this PR focused, please?

It seems that both `ColorBuilder` and `ColorDefinitionBuilder` are introduced in this PR so I don't understand so I don't agree with this.


I was asking for`class ColorBuilder` to be in `color-builder.ts` and not `class ColorDefinitionBuilder`.